Sofa/Component/MechanicalLoad/src/sofa/component/mechanicalload/NodalLinearDampingForceField.inl

```diff
@@ -0,0 +1,349 @@
+        }
+    }
+    else
+    {
+        const auto& coefs = d_dampingCoefficientVector.getValue();
+        std::size_t nbDampingCoeff = coefs.size();
+
+        if (nbDampingCoeff && bfactor)
+        {
+
+            for (std::size_t i = 0; i < dx.size(); i++)
+            {
+                for (unsigned j = 0; j < Deriv::total_size; j++)
+                {
+                    df[i][j] -= dx[i][j] * coefs[i][j] * bfactor;
+                }
+            }
+        }
+    }
+}
+
+template <class DataTypes>
+void NodalLinearDampingForceField<DataTypes>::buildStiffnessMatrix(core::behavior::StiffnessMatrix*)
+{
+    // NodalLinearDampingForceField is a pure damping component
+    // No stiffness is computed
+}
+
+template<class DataTypes>
+void NodalLinearDampingForceField<DataTypes>::addBToMatrix(sofa::linearalgebra::BaseMatrix * mat, SReal bFact, unsigned int& offset)
+{
+    if(!this->isComponentStateValid())
+        return;
+    
+    // Constant and isotropic damping
+    if(isConstantIsotropic && bFact)
+    {
+        const Real factor = d_dampingCoefficient.getValue() * bFact;
+        const unsigned int size = this->mstate->getSize();
+
+        for (std::size_t i = 0; i < size; i++)
+        {
+            const unsigned blockrow = offset + i * Deriv::total_size;
+            for (unsigned j = 0; j < Deriv::total_size; j++)
+            {
+                unsigned row = blockrow + j;
+                mat->add(row, row, -factor);
+            }
+        }
+    }
+    else
+    {
+        const auto& coefs = d_dampingCoefficientVector.getValue();
+        const std::size_t nbDampingCoeff = coefs.size();
+
+        if (nbDampingCoeff && bFact)
+        {
+            const unsigned int size = this->mstate->getSize();
+            for (std::size_t i = 0; i < size; i++)
+            {
+                const unsigned blockrow = offset + i * Deriv::total_size;
+                for (unsigned j = 0; j < Deriv::total_size; j++)
+                {
+                    unsigned row = blockrow + j;
+                    mat->add(row, row, -coefs[i][j] * bFact);
+                }
+            }
+        }
+    }
+}
+
+template <class DataTypes>
+void NodalLinearDampingForceField<DataTypes>::buildDampingMatrix(core::behavior::DampingMatrix* matrix)
+{
+    if(!this->isComponentStateValid())
+        return;
+    
+    auto dfdv = matrix->getForceDerivativeIn(this->mstate)
+        .withRespectToVelocityIn(this->mstate);
+
+    const unsigned int size = this->mstate->getSize();
+
+    // Constant and isotropic damping
+    if(isConstantIsotropic)
+    {
+        const Real dampingCoefficient = d_dampingCoefficient.getValue();
+
+        for (std::size_t i = 0; i < size; i++)
+        {
+            const unsigned blockrow = i * Deriv::total_size;
+            for (unsigned j = 0; j < Deriv::total_size; j++)
+            {
+                const unsigned row = blockrow + j;
+                dfdv(row, row) += -dampingCoefficient;
+            }
```

Comment:
You could probably add by blocks for more efficiency. In the case where `!isConstantIsotropic` too.

Reply:
https://github.com/sofa-framework/sofa/blob/5ceafe071fa804ff1230ee24706b4fce45621539/Sofa/Component/SolidMechanics/FEM/Elastic/src/sofa/component/solidmechanics/fem/elastic/HexahedronFEMForceField.inl#L1177

Does it help?